### ISSUE TITLE
Non intuitive behaviour of Tokenizer

### ISSUE THREAD

**Issue by hardikudeshi** [neuml/txtai#61]
Tokenizer fails in a way that is not intuitive to the user. Pls refer example below

```python
from txtai.tokenizer import Tokenizer

Tokenizer.tokenize("abc")
# output -> ["abc"]
Tokenizer.tokenize("12c")
# output -> []
Tokenizer.tokenize("c12")
# output -> ["c12"]
```
Looking at the source comments one would believe that `12c` will be tokenized and matched.
```python
# Require strings to be at least 2 characters long.
# Require at least 1 alpha character in string.
```
At the least, user should be warned that tokenizer did not match the string & hence returned a blank.

Possible rectification:
`\d{0,1}\d*[a-z]([\-.0-9:_a-z]{1,})*$`

The assumptions of Tokenizer should also be clearer in the main documentation.

**Comment by davidmezzetti** [comment]
Thank you for reporting this! I've corrected the documentation in the source code and added a note regarding the tokenizer to the README. A couple of similar issues (#39 and #43) were similar in nature regarding the tokenizer but I failed to update the documentation.

While the tokenizer is enabled by default, it is designed to work against standard English text. In other cases, it's best to disable the tokenizer (non-English transformer models) or use a custom tokenizer (for word embedding models).
